src/main/haskell/kore/src/Kore/Step/BaseStep.hs

```diff
@@ -182,8 +181,8 @@ stepWithAxiom
     -> AxiomPattern level
     -- ^ Rewriting axiom
     -> ExceptT
-        (Counter (StepError level Variable))
-        Counter
+        (Simplifier (StepError level Variable))
```

Comment:
I don't expect you to change this, but I want to make the comment publicly that I made to you privately: This should be `ExceptT (StepError level Variable) Simplifier (CommonExpandedPattern level, StepProof level)`. Having the error type be a nested `Simplifier` (or `Counter`) sets us up for all kinds of confusion.